### PR TITLE
Telemetry overview and alerting documentation rewrite

### PR DESCRIPTION
**Pull request by sewatson**

### CODE REVIEW

File: portal-sdk/templates/top-telemetry-alerting.md [hunk 1]
```diff
@@ -0,0 +1,707 @@
...
+            ...
+        }
+        ...
+    ]
+    ...
+}
+```
+
+[alerting-onboarding]: https://aka.ms/portalfx/alerting-onboarding
+[alerting-kusto-partner]: https://ailoganalyticsportal-privatecluster.cloudapp.net/clusters/azportal.kusto.windows.net/databases/Partner?q=H4sIAAAAAAAEAEvOKS0uSS3SUHesKsgvKknMUdfUS0ksSUxKLE7VUApILCrJSy1S0tRzSU1LLM0pcS7KBKrOTNTQBABHZQn9OQAAAA%3d%3d
```
[the remark this answers is not in the document]
Comment (sewatson) [reply portal-sdk/templates/top-telemetry-alerting.md]: https://dataexplorer.azure.com/clusters/azportalpartner/databases/Phms?query=H4sIAAAAAAAAA3NJTUsszSlxLsosSS3KTNTQBABwyXOmEQAAAA== is the link that they produce?

[the remark this answers is not in the document]
Comment (dbrankin) [reply portal-sdk/templates/top-telemetry-alerting.md]: That looks fine to me, but it doesn't resolve the DefaultCriteria function not being found. I'll prod Hua when he's free (he's working with someone at his desk) and ask him about it.

File: portal-sdk/templates/top-telemetry.md [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+
+The framework offers a common telemetry solution which as you develop and deploy your extension will track any common actions users take throughout your experience. If you wish to track specific actions in a custom manner that is also possible.
+
+The common set of actions is viewable via our PowerBi reports and the Kusto explorer (a query editor). If you chose to log your extension telemetry to the framework's tables then that will also be viewable via the Kusto explorer.
+
+Ask questions on: [https://stackoverflow.microsoft.com/questions/tagged?tagnames=ibiza-telemetry](https://stackoverflow.microsoft.com/questions/tagged?tagnames=ibiza-telemetry)
+
+## Viewing telemetry
+
+To view the telemetry which is logged throughout the portal you will first need permissions, then you will gain access to our PowerBi reports and the Kusto cluster for custom queries.
```
[the remark this answers is not in the document]
Comment (sewatson) [reply portal-sdk/templates/top-telemetry.md]: Fixed

File: portal-sdk/templates/top-telemetry.md [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+For all other questions please reach out to [Ibiza Telemetry](mailto:ibiza-telemetry@microsoft.com).
+
+#### Programmatic access
+
+We're currently not onboarding any more users on to programmatic access. We're in the process of making some infrastructure changes to support this ask.
+We'll update the documentation when we can onboard programmatic access.
+
+Kusto.Explorer: [Application](http://kusto-us/ke/Kusto.Explorer.application)
+
+### PowerBi reports
```
[the remark this answers is not in the document]
Comment (sewatson) [reply portal-sdk/templates/top-telemetry.md]: Yup did that on the first review 👍

File: portal-sdk/templates/top-telemetry.md [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+Name: Azportalpartner
+Data Source: [https://azportalpartner.kusto.windows.net](https://azportalpartner.kusto.windows.net)
+
+#### Supported databases
+
+Our Kusto cluster, [https://azportalpartner.kusto.windows.net](https://azportalpartner.kusto.windows.net), contains the following databases:
+
+| Name | Details |
+| ---- | ------- |
+| AzPtlCosmos | This is our main telemetry database. Data here is deduped, geo-coded, expanded and **filtered**. All the official dashboards\reports are based on this table. It is highly encouraged that this database is used for your needs. Data here is persisted for 30 days and excludes test traffic. |
```
[the remark this answers is not in the document]
Comment (sewatson) [reply portal-sdk/templates/top-telemetry.md]: We can always update this documentation with further changes as we complete the migration.

Updated the current documentation to reflect the timing better

File: portal-sdk/templates/top-telemetry.md [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+| Lens | This field gives the Lens name associated with the particular Action. This is derived from either then Name field or the Source field depending on the Action |
+| Name | The Name field usually changes it's format based on the Action. This field is usually used to identify the extension\Blade\Lens\Part associated with a particular Action. In most scenarios, it usually has the following format `Extension/<extensionName>/Blade/<BladeName>/Lens/<LensName>/PartInstance/<PartName>` |
+| SessionId | This represents each sessions that the user opens. SessionId refreshes everytime a user logs in\refreshes. |
+| Part | This field gives the Part name associated with the particular Action. This is derived from either then Name field or the Source field depending on the Action |
+| PreciseTimeStamp | This field gives the time the event was logged by the server. It is in UTC. |
+| TenantId | This field identifies the tenant the current users belongs to. |
+| UserId | This field identifies a user by PUID. We can use this to identify queries like daily active users, unique users using my feature, etc. |
```
[the remark this answers is not in the document]
Comment (sewatson) [reply portal-sdk/templates/top-telemetry.md]: Thank you.

PS you're typo threw me at first

File: portal-sdk/templates/top-telemetry.md [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+| UserAgent | This represents the user agent of the user. This is a standard UserAgentString - [User Agent](https://en.wikipedia.org/wiki/User_agent) |
```
[the remark this answers is not in the document]
Comment (sewatson) [reply portal-sdk/templates/top-telemetry.md]: Done.

File: portal-sdk/templates/top-telemetry.md [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+##### ClientTelmetry (AzurePortal)
+
+> Note the casing difference between AzurePortal and AzPtlCosmos
+
+| Field Name | Details |
+| ---------- | ------- |
+| acceptLanguage | This field indicates the language and locale that the user's browser is emitting. |
+| action | This represents an event in the Portal. |
+| actionModifier | This is used in tandem with the Action field. This represents a status of a particular Action. So for BladeReady for eg., you will see ActionModifier values of start, complete & cancel |
+| browserId | This is an identify which represents a given instance of the users browser, a single id may span multiple users if they switch accounts. |
```
[the remark this answers is not in the document]
Comment (sewatson) [reply portal-sdk/templates/top-telemetry.md]: Sounds good. Done

[the remark this answers is not in the document]
Comment (sewatson) [reply portal-sdk/templates/top-telemetry.md]: Done

File: portal-sdk/templates/top-telemetry.md [hunk 8]
```diff
@@ -0,0 +1,346 @@
...
+| clientTime | This field gives the actual time of the event according to the client's clock (which can be off based on the client settings). This is a good field to reconstruct the precise sequence of events. **This is in ticks, you will need to convert it to a datetime object for easy reading** |
+| clientVersion | This is the version which the user's client was running when they logged the telemetry event. |
+| data | The Data field is the most dynamic field in telemetry. It is a JSON object with no set structure. They often contain information specific to a particular Action.  |
+| duration | This field gives the duration a particular Action took to complete. This value is non-zero only for Actions with ActionModifier having values either "complete", "succeeded", etc. The time is in milliseconds. |
+| journeyId | This field provides the journey Id for each action. A journey is basically a tiny sub-session within which a user navigates a flow of blades. This id allows us to identify the actions that the user took within any given journey, etc... |
```
[the remark this answers is not in the document]
Comment (sewatson) [reply portal-sdk/templates/top-telemetry.md]: Already done with the first nit'ng :).

File: portal-sdk/templates/top-telemetry.md [hunk 9]
```diff
@@ -0,0 +1,346 @@
...
+| ---------- | ------- |
+| extension | Name of the extension who logged the particular telemetry event |
+
+#### Tracked Actions
+
+| Category | Action | Details |
+| -------- | ------ | ------- |
+| Blade | BladeLoadErrored | Triggered when loading a blade failed. **This event is used to track blade errors in our reliability metrics.** |
+| Blade | BladeFullReady | The blade's constructor, onInitialize/onInputsSet have resolved and if the blade contains any parts, all part's constructor and onInputsSet have also resolved. |
+| Blade | BladeButtonClicked | When the pin, unpin, maximize, minimize or close button on a blade is clicked. |
```
[the remark this answers is not in the document]
Comment (sewatson) [reply portal-sdk/templates/top-telemetry.md]: Removed

File: portal-sdk/generated/top-telemetry-alerting.md [hunk 10]
```diff
@@ -0,0 +1,797 @@
...
+Once an alert is triggered it will automatically open an IcM on the owning team.
+
+<a name="overview-what-alerting-is-available"></a>
+## What alerting is available
+
+There are number of framework provided alerts:
+
+1. Extension SDK age
+    - Sev3 IcM incdient for an extension when its SDK is older than 60 days
+    - Sev2 for over 90 days
```
Comment (adamabmsft) [inline portal-sdk/generated/top-telemetry-alerting.md]: 120 days now, right?

Comment (sewatson) [reply portal-sdk/generated/top-telemetry-alerting.md]: I believe we still send sev 2's if they are within 30 days of the SLA.

Comment (sewatson) [reply portal-sdk/generated/top-telemetry-alerting.md]: This PR is also completed.

**Pull request merged**